### PR TITLE
fix: make set token multiplier optional

### PR DESCRIPTION
**Pull request by ischasny**
For backwards compatibility make setting token multiplier on L1 optional. If private key for token multiplier setter is not provided, L1 quote update will be skipped.

### CODE REVIEW

File: core/node/base_token_adjuster/src/base_token_ratio_persister.rs [hunk 1]
```diff
@@ -90,53 +83,63 @@ impl BaseTokenRatioPersister {
         let new_ratio = self.retry_fetch_ratio().await?;
         self.persist_ratio(new_ratio).await?;
 
-        let max_attempts = self.config.l1_tx_sending_max_attempts;
-        let sleep_duration = self.config.l1_tx_sending_sleep_duration();
-        let mut result: anyhow::Result<()> = Ok(());
-        let mut prev_base_fee_per_gas: Option<u64> = None;
-        let mut prev_priority_fee_per_gas: Option<u64> = None;
+        if let Some(l1_params) = &self.l1_params {
```
Comment (perekopskiy) [inline core/node/base_token_adjuster/src/base_token_ratio_persister.rs:86]: nit:
`let Some(l1_params) = &self.l1_params else { return Ok(()) }`
will look nicely

File: core/node/node_framework/src/implementations/layers/base_token/base_token_ratio_persister.rs [hunk 2]
```diff
@@ -81,37 +82,37 @@ impl WiringLayer for BaseTokenRatioPersisterLayer {
...
-        let token_multiplier_setter_wallet = self
-            .wallets_config
-            .token_multiplier_setter
-            .expect("base token adjuster wallet is not set")
-            .wallet;
 
-        let tms_private_key = token_multiplier_setter_wallet.private_key();
-        let tms_address = token_multiplier_setter_wallet.address();
-        let EthInterfaceResource(query_client) = input.eth_client;
+        let mut l1_params: Option<BaseTokenRatioPersisterL1Params> = None;
```
Comment (perekopskiy) [inline core/node/node_framework/src/implementations/layers/base_token/base_token_ratio_persister.rs:86]: you can avoid mutability with `let l1_params  = self.wallets_config.token_multiplier_setter.map(...)`

File: core/node/base_token_adjuster/src/base_token_ratio_persister.rs (hunk 1 above)
Comment (ischasny) [reply core/node/base_token_adjuster/src/base_token_ratio_persister.rs:86]: It's much nicer indeed - fixed

File: core/node/node_framework/src/implementations/layers/base_token/base_token_ratio_persister.rs (hunk 2 above)
Comment (ischasny) [reply core/node/node_framework/src/implementations/layers/base_token/base_token_ratio_persister.rs:86]: thank you - fixed

**Review by perekopskiy — APPROVED** [review]

**Pull request merged**